packages/aserto/src/aserto/client/directory/__init__.py

```diff
@@ -0,0 +1,450 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple, TypedDict
+
+import grpc
+from aserto.directory.common.v2 import (
+    Object,
+    ObjectIdentifier,
+    ObjectTypeIdentifier,
+    PaginationRequest,
+    PaginationResponse,
+    PermissionIdentifier,
+    Relation,
+    RelationIdentifier,
+    RelationTypeIdentifier,
+)
+from aserto.directory.exporter.v2 import ExporterStub
+from aserto.directory.importer.v2 import ImporterStub
+from aserto.directory.reader.v2 import (
+    CheckPermissionRequest,
+    CheckRelationRequest,
+    GetObjectManyRequest,
+    GetObjectRequest,
+    GetObjectsRequest,
+    GetObjectsResponse,
+    GetRelationRequest,
+    GetRelationsRequest,
+    GetRelationsResponse,
+    ReaderStub,
+)
+from aserto.directory.writer.v2 import (
+    DeleteObjectRequest,
+    DeleteRelationRequest,
+    SetObjectRequest,
+    SetRelationRequest,
+    WriterStub,
+)
+
+
+class NotFoundError(Exception):
+    pass
+
+
+class Directory:
+    def __init__(self, *, address: str, api_key: str, tenant_id: str, ca_cert: str) -> None:
+        self._channel = grpc.secure_channel(
+            target=address, credentials=self._channel_credentials(cert=ca_cert)
+        )
+        self._metadata = self._get_metadata(api_key=api_key, tenant_id=tenant_id)
+        self.reader = ReaderStub(self._channel)
+        self.writer = WriterStub(self._channel)
+        self.importer = ImporterStub(self._channel)
+        self.exporter = ExporterStub(self._channel)
+
```

Comment:
I think it would be good to make `Directory` a context manager and close the channel when exiting the context. 
```python
def __enter__(self):
    return self

def __exit__(self, type, value, traceback):
    self.close_channel()
```

This will allow callers to use `Directory` in `with` statements like:
```python
with Directory("<address>", "<key>", "<tenant_id">) as directory:
    # do stuff with the directory
    objects = directory.get_objects("user")

# at this point the directory is discarded and the channel closed.
```